Bump Diffusers Dependency

hipsterusername opened the pull request:
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ X ] Optimization
- [ ] Documentation Update
- [ ] Community Node Submission


## Have you discussed this change with the InvokeAI team?
- [ ] Yes
- [ X ] No, because: dependency bump

      
## Have you updated all relevant documentation?
- [ ] Yes
- [ x ] No


## Description
Updating diffusers to .24 - fixes a few issues. Needs to be tested to ensure things like our IP Adapter implementation don't break

Millu commented:
Works for me! Tested with new & existing venvs

gogurtenjoyer commented:
Howdy! With diffusers 0.24.0 and huggingface-hub 0.19.4, I get the following when trying to generate anything:

```
[2023-12-09 16:07:59,538]::[InvokeAI]::ERROR --> Traceback (most recent call last):
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/services/invocation_processor/invocation_processor_default.py", line 104, in __process
    outputs = invocation.invoke_internal(
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/baseinvocation.py", line 591, in invoke_internal
    output = self.invoke(context)
             ^^^^^^^^^^^^^^^^^^^^
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/latent.py", line 730, in invoke
    pipeline = self.create_pipeline(unet, scheduler)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/latent.py", line 371, in create_pipeline
    return StableDiffusionGeneratorPipeline(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/backend/stable_diffusion/diffusers_pipeline.py", line 244, in __init__
    super().__init__(
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py", line 236, in __init__
    self.register_modules(
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/diffusers/pipelines/pipeline_utils.py", line 566, in register_modules
    library = not_compiled_module.__module__.split(".")[0]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'bool' object has no attribute '__module__'

[2023-12-09 16:07:59,543]::[InvokeAI]::ERROR --> Error while invoking:
'bool' object has no attribute '__module__'
[2023-12-09 16:07:59,734]::[InvokeAI]::ERROR --> Traceback (most recent call last):
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/services/invocation_processor/invocation_processor_default.py", line 104, in __process
    outputs = invocation.invoke_internal(
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/baseinvocation.py", line 591, in invoke_internal
    output = self.invoke(context)
             ^^^^^^^^^^^^^^^^^^^^
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/latent.py", line 730, in invoke
    pipeline = self.create_pipeline(unet, scheduler)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/latent.py", line 371, in create_pipeline
    return StableDiffusionGeneratorPipeline(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/backend/stable_diffusion/diffusers_pipeline.py", line 244, in __init__
    super().__init__(
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py", line 236, in __init__
    self.register_modules(
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/diffusers/pipelines/pipeline_utils.py", line 566, in register_modules
    library = not_compiled_module.__module__.split(".")[0]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'bool' object has no attribute '__module__'

[2023-12-09 16:07:59,735]::[InvokeAI]::ERROR --> Error while invoking:
'bool' object has no attribute '__module__'
[2023-12-09 16:07:59,922]::[InvokeAI]::ERROR --> Traceback (most recent call last):
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/services/invocation_processor/invocation_processor_default.py", line 104, in __process
    outputs = invocation.invoke_internal(
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/baseinvocation.py", line 591, in invoke_internal
    output = self.invoke(context)
             ^^^^^^^^^^^^^^^^^^^^
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/latent.py", line 730, in invoke
    pipeline = self.create_pipeline(unet, scheduler)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/latent.py", line 371, in create_pipeline
    return StableDiffusionGeneratorPipeline(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/backend/stable_diffusion/diffusers_pipeline.py", line 244, in __init__
    super().__init__(
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py", line 236, in __init__
    self.register_modules(
  File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/diffusers/pipelines/pipeline_utils.py", line 566, in register_modules
    library = not_compiled_module.__module__.split(".")[0]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'bool' object has no attribute '__module__'

[2023-12-09 16:07:59,923]::[InvokeAI]::ERROR --> Error while invoking:
'bool' object has no attribute '__module__'
```

hipsterusername commented:
> Howdy! With diffusers 0.24.0 and huggingface-hub 0.19.4, I get the following when trying to generate anything:
> 
> ```
> [2023-12-09 16:07:59,538]::[InvokeAI]::ERROR --> Traceback (most recent call last):
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/services/invocation_processor/invocation_processor_default.py", line 104, in __process
>     outputs = invocation.invoke_internal(
>               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/baseinvocation.py", line 591, in invoke_internal
>     output = self.invoke(context)
>              ^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
>     return func(*args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/latent.py", line 730, in invoke
>     pipeline = self.create_pipeline(unet, scheduler)
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/latent.py", line 371, in create_pipeline
>     return StableDiffusionGeneratorPipeline(
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/backend/stable_diffusion/diffusers_pipeline.py", line 244, in __init__
>     super().__init__(
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py", line 236, in __init__
>     self.register_modules(
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/diffusers/pipelines/pipeline_utils.py", line 566, in register_modules
>     library = not_compiled_module.__module__.split(".")[0]
>               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> AttributeError: 'bool' object has no attribute '__module__'
> 
> [2023-12-09 16:07:59,543]::[InvokeAI]::ERROR --> Error while invoking:
> 'bool' object has no attribute '__module__'
> [2023-12-09 16:07:59,734]::[InvokeAI]::ERROR --> Traceback (most recent call last):
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/services/invocation_processor/invocation_processor_default.py", line 104, in __process
>     outputs = invocation.invoke_internal(
>               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/baseinvocation.py", line 591, in invoke_internal
>     output = self.invoke(context)
>              ^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
>     return func(*args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/latent.py", line 730, in invoke
>     pipeline = self.create_pipeline(unet, scheduler)
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/latent.py", line 371, in create_pipeline
>     return StableDiffusionGeneratorPipeline(
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/backend/stable_diffusion/diffusers_pipeline.py", line 244, in __init__
>     super().__init__(
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py", line 236, in __init__
>     self.register_modules(
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/diffusers/pipelines/pipeline_utils.py", line 566, in register_modules
>     library = not_compiled_module.__module__.split(".")[0]
>               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> AttributeError: 'bool' object has no attribute '__module__'
> 
> [2023-12-09 16:07:59,735]::[InvokeAI]::ERROR --> Error while invoking:
> 'bool' object has no attribute '__module__'
> [2023-12-09 16:07:59,922]::[InvokeAI]::ERROR --> Traceback (most recent call last):
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/services/invocation_processor/invocation_processor_default.py", line 104, in __process
>     outputs = invocation.invoke_internal(
>               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/baseinvocation.py", line 591, in invoke_internal
>     output = self.invoke(context)
>              ^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
>     return func(*args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/latent.py", line 730, in invoke
>     pipeline = self.create_pipeline(unet, scheduler)
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/latent.py", line 371, in create_pipeline
>     return StableDiffusionGeneratorPipeline(
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/backend/stable_diffusion/diffusers_pipeline.py", line 244, in __init__
>     super().__init__(
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py", line 236, in __init__
>     self.register_modules(
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/diffusers/pipelines/pipeline_utils.py", line 566, in register_modules
>     library = not_compiled_module.__module__.split(".")[0]
>               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> AttributeError: 'bool' object has no attribute '__module__'
> 
> [2023-12-09 16:07:59,923]::[InvokeAI]::ERROR --> Error while invoking:
> 'bool' object has no attribute '__module__'
> ```



> Howdy! With diffusers 0.24.0 and huggingface-hub 0.19.4, I get the following when trying to generate anything:
> 
> ```
> [2023-12-09 16:07:59,538]::[InvokeAI]::ERROR --> Traceback (most recent call last):
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/services/invocation_processor/invocation_processor_default.py", line 104, in __process
>     outputs = invocation.invoke_internal(
>               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/baseinvocation.py", line 591, in invoke_internal
>     output = self.invoke(context)
>              ^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
>     return func(*args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/latent.py", line 730, in invoke
>     pipeline = self.create_pipeline(unet, scheduler)
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/latent.py", line 371, in create_pipeline
>     return StableDiffusionGeneratorPipeline(
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/backend/stable_diffusion/diffusers_pipeline.py", line 244, in __init__
>     super().__init__(
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py", line 236, in __init__
>     self.register_modules(
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/diffusers/pipelines/pipeline_utils.py", line 566, in register_modules
>     library = not_compiled_module.__module__.split(".")[0]
>               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> AttributeError: 'bool' object has no attribute '__module__'
> 
> [2023-12-09 16:07:59,543]::[InvokeAI]::ERROR --> Error while invoking:
> 'bool' object has no attribute '__module__'
> [2023-12-09 16:07:59,734]::[InvokeAI]::ERROR --> Traceback (most recent call last):
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/services/invocation_processor/invocation_processor_default.py", line 104, in __process
>     outputs = invocation.invoke_internal(
>               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/baseinvocation.py", line 591, in invoke_internal
>     output = self.invoke(context)
>              ^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
>     return func(*args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/latent.py", line 730, in invoke
>     pipeline = self.create_pipeline(unet, scheduler)
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/latent.py", line 371, in create_pipeline
>     return StableDiffusionGeneratorPipeline(
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/backend/stable_diffusion/diffusers_pipeline.py", line 244, in __init__
>     super().__init__(
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py", line 236, in __init__
>     self.register_modules(
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/diffusers/pipelines/pipeline_utils.py", line 566, in register_modules
>     library = not_compiled_module.__module__.split(".")[0]
>               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> AttributeError: 'bool' object has no attribute '__module__'
> 
> [2023-12-09 16:07:59,735]::[InvokeAI]::ERROR --> Error while invoking:
> 'bool' object has no attribute '__module__'
> [2023-12-09 16:07:59,922]::[InvokeAI]::ERROR --> Traceback (most recent call last):
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/services/invocation_processor/invocation_processor_default.py", line 104, in __process
>     outputs = invocation.invoke_internal(
>               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/baseinvocation.py", line 591, in invoke_internal
>     output = self.invoke(context)
>              ^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
>     return func(*args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/latent.py", line 730, in invoke
>     pipeline = self.create_pipeline(unet, scheduler)
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/app/invocations/latent.py", line 371, in create_pipeline
>     return StableDiffusionGeneratorPipeline(
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/invokeai/backend/stable_diffusion/diffusers_pipeline.py", line 244, in __init__
>     super().__init__(
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py", line 236, in __init__
>     self.register_modules(
>   File "/Users/gogurtdisliker/invokeai/.venv/lib/python3.11/site-packages/diffusers/pipelines/pipeline_utils.py", line 566, in register_modules
>     library = not_compiled_module.__module__.split(".")[0]
>               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> AttributeError: 'bool' object has no attribute '__module__'
> 
> [2023-12-09 16:07:59,923]::[InvokeAI]::ERROR --> Error while invoking:
> 'bool' object has no attribute '__module__'
> ```

Actually getting this as well (I must not have tested what I thought I did...)

Digging into it.

RyanJDick commented:
Should be fixed in [f2d611c](https://github.com/invoke-ai/InvokeAI/pull/5243/commits/f2d611c0ede26dc67cad43659850a5351dffd28f).

The interface to `StableDiffusionPipeline` changed here: https://github.com/huggingface/diffusers/blame/1d686bac8146037e97f3fd8c56e4063230f71751/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L167

We pass kwargs now instead of positional args to be more robust to future changes.

I also removed the call to `self.register_modules(...)`. It's done in the super() constructor, and I couldn't see a good reason to do it again. @lstein let me know if you can think of an issue with this.

RyanJDick commented:
It's worth noting that the change that broke our pipeline was diffusers adding native IP-Adapter support to `StableDiffusionPipeline`. Might be worth considering switching to their implementation at some point, but ours still seems to be working for now.
